### PR TITLE
test(ApplicationShell): reflect the changes introduced in the new navbar

### PR DESCRIPTION
**Pull request by kark**
#### Summary

This PR includes test coverage for the new Navbar functionality.

### CODE REVIEW

File: packages/application-shell/src/components/navbar/navbar-new/menu-items.tsx [hunk 1]
```diff
@@ -212,7 +212,7 @@ const MenuGroup = forwardRef<HTMLUListElement, MenuGroupProps>((props, ref) => {
     props.level === 2 && props.isActive && !props.isExpanded;
   return (
     <ul
-      ref={ref}
+      ref={ref && props.level === 2 ? ref : null}
```
Comment (kark) [inline packages/application-shell/src/components/navbar/navbar-new/menu-items.tsx:215]: We don't need to pass refs to the level 1 submenu

File: packages/application-shell/src/components/application-shell/application-shell.spec.js [hunk 2]
```diff
@@ -1372,3 +1381,1205 @@ describe('when navbar menu items do not match given data fences', () => {
     });
   });
 });
+describe('With new navbar', () => {
```
Comment (emmenko) [inline packages/application-shell/src/components/application-shell/application-shell.spec.js:1384]: Is it on purpose to duplicate all the tests? 🤔

File: packages/application-shell/src/components/application-shell/application-shell.spec.js [hunk 3]
```diff
@@ -1372,3 +1381,1205 @@ describe('when navbar menu items do not match given data fences', () => {
...
+      const submenuLink = within(container)
+        .getByText(mainSubmenuLabel.value)
+        // eslint-disable-next-line testing-library/no-node-access
+        .closest('a');
+
+      // Go to the link
+      fireEvent.click(submenuLink);
+
+      // Ensure that the link becomes active
+      expect(submenuLink).toHaveAttribute('aria-current', 'page');
```
[the remark this answers is not in the document]
Comment (emmenko) [reply packages/application-shell/src/components/application-shell/application-shell.spec.js:2075]: Thanks for the context. Should we consider then to take some technical debts to improve the accessibility / testing part?

[the remark this answers is not in the document]
Comment (kark) [reply packages/application-shell/src/components/application-shell/application-shell.spec.js]: Yes, thanks, seems like a good idea. I added a TODO comment.

File: packages/application-shell/src/components/navbar/navbar-new/navbar.tsx [hunk 4]
```diff
@@ -246,6 +246,7 @@ export const ApplicationMenu = (props: ApplicationMenuProps) => {
               <div
                 className={styles['tooltip']}
                 aria-owns={`group-${props.menu.key}`}
+                data-testid="tooltip"
```
[the remark this answers is not in the document]
Comment (kark) [reply packages/application-shell/src/components/navbar/navbar-new/navbar.tsx]: Thank you, that's definitely not needed. Changed in a48a6428a67db9d0fbd42bfb653e448861a60714

File: packages/application-shell/src/components/application-shell/application-shell.spec.js (hunk 2 above)
Comment (kark) [reply packages/application-shell/src/components/application-shell/application-shell.spec.js:1384]: Upon second thought, I removed the tests that are unrelated to the changes we're making.

**Review by chloe0592 — APPROVED** [review]
Was trying to think of more required tests but I think you've covered it all. Great work! Thank you! 🙇‍♀️

File: packages/application-shell/src/components/application-shell/application-shell.spec.js (hunk 2 above)
Comment (emmenko) [reply packages/application-shell/src/components/application-shell/application-shell.spec.js:1384]: Thanks!

**Review by emmenko — APPROVED** [review]
🚀

**Pull request merged**